testSuite/shared/src/test/scala/testsuite/javalib/util/CurrencyTest.scala

```diff
@@ -0,0 +1,281 @@
+    ),
+    CurrencyTest(
+      testByLocale = Locale.GERMANY,
+      testByLanguageTag = "de-DE",
+      jsResults  = CurrencyTestResults("EUR", 978, 2, "EUR", "欧元"),
+      jvmResults = CurrencyTestResults("EUR", 978, 2, "€", "欧元")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.ITALY,
+      testByLanguageTag = "it-IT",
+      jsResults  = CurrencyTestResults("EUR", 978, 2, "EUR", "欧元"),
+      jvmResults = CurrencyTestResults("EUR", 978, 2, "€", "欧元")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.JAPAN,
+      testByLanguageTag = "ja-JP",
+      jsResults  = CurrencyTestResults("JPY", 392, 0, "JPY", "日元"),
+      jvmResults = CurrencyTestResults("JPY", 392, 0, "JP¥", "日元")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.KOREA,
+      testByLanguageTag = "ko-KR",
+      jsResults  = CurrencyTestResults("KRW", 410, 0, "KRW", "韩元"),
+      jvmResults = CurrencyTestResults("KRW", 410, 0, "￦", "韩圆")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.PRC,
+      testByLanguageTag = "zh-CN",
+      jsResults  = CurrencyTestResults("CNY", 156, 2, "\u00A5"/* narrow width yen ¥ */, "人民币"),
+      jvmResults = CurrencyTestResults("CNY", 156, 2, "￥", "人民币")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.TAIWAN,
+      testByLanguageTag = "zh-TW",
+      jsResults  = CurrencyTestResults("TWD", 901, 2, "TWD", "新台币"),
+      jvmResults = CurrencyTestResults("TWD", 901, 2, "NT$", "新台币")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.UK,
+      testByLanguageTag = "en-GB",
+      jsResults  = CurrencyTestResults("GBP", 826, 2, "GBP", "英镑"),
+      jvmResults = CurrencyTestResults("GBP", 826, 2, "£", "英镑")
+    ),
+    CurrencyTest(
+      testByLocale = Locale.US,
+      testByLanguageTag = "en-US",
+      jsResults  = CurrencyTestResults("USD", 840, 2, "USD", "美元"),
+      jvmResults = CurrencyTestResults("USD", 840, 2, "US$", "美元")
+    )
+  ))
+
+  private val countryTests = Seq(
+    US_standardCountryLocales,
+    GERMANY_standardCountryLocales,
+    CHINA_standardCountryLocales
+  )
+
+  private def testCurrency(currency: Currency, test: CurrencyTest, expectedResults: CurrencyTestResults): Unit = {
+    import expectedResults._
+
+    val currentTestMsg: String = s"currency: ${currency} and test: $test and expectedResutls: $expectedResults"
+    assertEquals(currentTestMsg, expectedCurrencyCode, currency.getCurrencyCode)
+    assertEquals(currentTestMsg, expectedNumericCode, currency.getNumericCode)
+    assertEquals(currentTestMsg, expectedFractionDigits, currency.getDefaultFractionDigits)
+    assertEquals(currentTestMsg, expectedSymbol, currency.getSymbol)
+    assertEquals(currentTestMsg, expectedDisplayName, currency.getDisplayName)
+  }
+
+  protected def test_standard_locales(f: CurrencyTest => CurrencyTestResults): Unit = {
+    countryTests.foreach{ regionTest: DefaultLocaleCurrencyTests =>
+      Locale.setDefault(regionTest.defaultLocale)
```

Comment:
Let me look into that again, I think what I was going for was to test for the different behavior for the output if the instance of Currency you have is different than the default locale, or if a currency is for a locale, but I'm not sure that got ported over to this current test...